shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -95,6 +95,24 @@ public Counter call() throws Exception {
     }
 
     @Override
+    public long getCounterValue(String name) {
+        Exceptions.checkNotNullOrEmpty(name, "name");
+        String counterName = name + ".Counter";
+        try {
+            Counter counter = countersCache.get(counterName, new Callable<Counter>() {
+                @Override
+                public Counter call() throws Exception {
+                    return underlying.createCounter(counterName);
+                }
+            });
+            return counter.get();
```

Comment:
Yes, we are storing the dynamic metrics' values in cache. If there is cache miss (value doesn't exist or value is not present due to cache eviction ) then we are resetting the counter.  Correct me if I am wrong @fpj .